server/fleet/microsoft_mdm.go

```diff
@@ -2,8 +2,214 @@ package fleet
 
 import (
 	"encoding/xml"
+	"errors"
+	"fmt"
+
+	mdm "github.com/fleetdm/fleet/v4/server/mdm/microsoft"
 )
 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////
+// MS-MDE2 XML types used by the SOAP protocol
+// MS-MDE2 is a client-to-server protocol that consists of a SOAP-based Web service.
+// SOAP is a lightweight and XML based protocol that consists of three parts:
+// - An envelope that defines a framework for describing what is in a message and how to process it
+// - A set of encoding rules for expressing instances of application-defined datatypes
+// - And a convention for representing remote procedure calls and responses.
+
+// SoapResponse is the Soap Envelope Response type for MS-MDE2 responses from the server
+// This envelope XML message is composed by a mandatory SOAP envelope, a SOAP header, and a SOAP body
+type SoapResponse struct {
+	XMLName xml.Name       `xml:"s:Envelope"`
+	XmlNSS  string         `xml:"xmlns:s,attr"`
```

Comment:
Nit: idiomatic Go is to use all uppercase for acronyms, and some linters enforce that. If we want to enable those eventually, it would help to conform to that rule (`XMLNSS`, etc., as is already the case for `XMLName`).

Reply:
Good catch! fixed